Source/Classes/AnimatedImages/PINGIFAnimatedImage.m

```diff
@@ -0,0 +1,151 @@
+//
+//  PINGIFAnimatedImage.m
+//  PINRemoteImage
+//
+//  Created by Garrett Moon on 9/17/17.
+//  Copyright © 2017 Pinterest. All rights reserved.
+//
+
+#import "PINGIFAnimatedImage.h"
+
+#import <ImageIO/ImageIO.h>
+#if PIN_TARGET_IOS
+#import <MobileCoreServices/UTCoreTypes.h>
+#elif PIN_TARGET_MAC
+#import <CoreServices/CoreServices.h>
+#endif
+
+#import "PINImage+DecodedImage.h"
+
+@interface PINGIFAnimatedImage ()
+{
+    NSData *_animatedImageData;
+    CGImageSourceRef _imageSource;
+    uint32_t _width;
+    uint32_t _height;
+    BOOL _hasAlpha;
+    size_t _frameCount;
+    size_t _loopCount;
+    CFTimeInterval *_durations;
+    NSError *_error;
+}
+@end
+
+@implementation PINGIFAnimatedImage
+
+- (instancetype)initWithAnimatedImageData:(NSData *)animatedImageData
+{
+    if (self = [super init]) {
+        _imageSource =
+            CGImageSourceCreateWithData((CFDataRef)animatedImageData,
+                                        (CFDictionaryRef)@{(__bridge NSString *)kCGImageSourceTypeIdentifierHint:
+                                                               (__bridge NSString *)kUTTypeGIF,
+                                                           (__bridge NSString *)kCGImageSourceShouldCache:
+                                                               (__bridge NSNumber *)kCFBooleanFalse});
+        if (_imageSource) {
+            _frameCount = (uint32_t)CGImageSourceGetCount(_imageSource);
+            NSDictionary *imageProperties = (__bridge_transfer NSDictionary *)CGImageSourceCopyProperties(_imageSource, nil);
+            _loopCount = (uint32_t)[[[imageProperties objectForKey:(__bridge NSString *)kCGImagePropertyGIFDictionary]
+                                     objectForKey:(__bridge NSString *)kCGImagePropertyGIFLoopCount] unsignedLongValue];
+            _durations = malloc(sizeof(CFTimeInterval) * _frameCount);
+            imageProperties = (__bridge_transfer NSDictionary *)
+                CGImageSourceCopyPropertiesAtIndex(_imageSource,
+                                                   0,
+                                                   (CFDictionaryRef)@{(__bridge NSString *)kCGImageSourceShouldCache:
+                                                                          (__bridge NSNumber *)kCFBooleanFalse});
+            _width = (uint32_t)[(NSNumber *)imageProperties[(__bridge NSString *)kCGImagePropertyPixelWidth] unsignedIntegerValue];
+            _height = (uint32_t)[(NSNumber *)imageProperties[(__bridge NSString *)kCGImagePropertyPixelHeight] unsignedIntegerValue];
```

Comment:
Not a blocker at the moment, but I've seen loading GIF loopCount and frame durations become a bottleneck on larger GIFs. I was working on loading loopCount and the durations asynchronously in https://github.com/timonus/FLAnimatedImage/commit/bddff94ba8f340dd31e2b5a122a6dae6867e9ea3 and https://github.com/timonus/FLAnimatedImage/commit/0198b74c30479b65b9dc8632e81c4426f1757a7c respectively, but never finished. Could be a nice thing to have here if we ever notice performance issues during initialization of GIFs. `playbackReady` could be adjusted to only become true once those are loaded.